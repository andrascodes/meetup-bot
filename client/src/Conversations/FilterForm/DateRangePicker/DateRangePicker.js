import React from 'react'

export const DateRangePicker = props => {

  const minStartDate = props.minStartDate || undefined
  const maxEndDate = props.maxEndDate || undefined
  const defaultStartValue = props.defaultStartValue || props.minStartDate || undefined
  const defaultEndValue = props.defaultEndValue || props.maxEndDate || undefined

  const maxStartDate = props.defaultEndValue || props.maxEndDate || undefined
  const minEndDate = props.defaultStartValue || props.minStartDate || undefined
  

  return (
    <div className="DateRangePicker">
      <label htmlFor="from" className="DatePickerLabel">Date between:</label>
      <input type="date" id="from" className="DateInput" 
        min={minStartDate}
        max={maxStartDate}
        onInput={props.handleStartInput}
        defaultValue={defaultStartValue}
      />
      <label htmlFor="to" className="DatePickerLabel">and</label>
      <input type="date" id="to" className="DateInput" 
        min={minEndDate}
        max={maxEndDate}
        onInput={props.handleEndInput}
        defaultValue={defaultEndValue}
      />
    </div>
  )
}

DateRangePicker.PropTypes = {
  minStartDate: React.PropTypes.string,
  maxEndDate: React.PropTypes.string,
  defaultStartValue: React.PropTypes.string,
  defaultEndValue: React.PropTypes.string,
  handleStartInput: React.PropTypes.func.isRequired,
  handleEndInput: React.PropTypes.func.isRequired
}