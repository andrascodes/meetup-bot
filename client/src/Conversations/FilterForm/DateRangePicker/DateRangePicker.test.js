import React from 'react'
import renderer from 'react-test-renderer'

import { DateRangePicker } from './DateRangePicker'

describe('DateRangePicker', () => {

  test('renders correctly', () => {
    
    const tree = renderer.create(
      <DateRangePicker 
        minStartDate={'2017-04-13'}
        maxEndDate={'2017-05-14'}
        defaultStartValue={'2017-04-15'}
        defaultEndValue={'2017-04-20'}
        handleStartInput={() => {}}
        handleEndInput={() => {}}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  
  test('renders correctly when no defaultValue is set', () => {
    
    const tree = renderer.create(
      <DateRangePicker 
        minStartDate={'2017-04-13'}
        maxEndDate={'2017-05-14'}
        handleStartInput={() => {}}
        handleEndInput={() => {}}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders correctly when no minDate is set', () => {
    
    const tree = renderer.create(
      <DateRangePicker 
        defaultStartValue={'2017-04-15'}
        defaultEndValue={'2017-04-20'}
        handleStartInput={() => {}}
        handleEndInput={() => {}}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders correctly when no minDate or defaultValue is set', () => {
    
    const tree = renderer.create(
      <DateRangePicker 
        handleStartInput={() => {}}
        handleEndInput={() => {}}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

})