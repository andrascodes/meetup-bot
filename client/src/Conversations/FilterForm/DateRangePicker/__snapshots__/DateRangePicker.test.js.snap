exports[`DateRangePicker renders correctly 1`] = `
<div
  className="DateRangePicker">
  <label
    className="DatePickerLabel"
    htmlFor="from">
    Date between:
  </label>
  <input
    className="DateInput"
    defaultValue="2017-04-15"
    id="from"
    max="2017-04-20"
    min="2017-04-13"
    onInput={[Function]}
    type="date" />
  <label
    className="DatePickerLabel"
    htmlFor="to">
    and
  </label>
  <input
    className="DateInput"
    defaultValue="2017-04-20"
    id="to"
    max="2017-05-14"
    min="2017-04-15"
    onInput={[Function]}
    type="date" />
</div>
`;

exports[`DateRangePicker renders correctly when no defaultValue is set 1`] = `
<div
  className="DateRangePicker">
  <label
    className="DatePickerLabel"
    htmlFor="from">
    Date between:
  </label>
  <input
    className="DateInput"
    defaultValue="2017-04-13"
    id="from"
    max="2017-05-14"
    min="2017-04-13"
    onInput={[Function]}
    type="date" />
  <label
    className="DatePickerLabel"
    htmlFor="to">
    and
  </label>
  <input
    className="DateInput"
    defaultValue="2017-05-14"
    id="to"
    max="2017-05-14"
    min="2017-04-13"
    onInput={[Function]}
    type="date" />
</div>
`;

exports[`DateRangePicker renders correctly when no minDate is set 1`] = `
<div
  className="DateRangePicker">
  <label
    className="DatePickerLabel"
    htmlFor="from">
    Date between:
  </label>
  <input
    className="DateInput"
    defaultValue="2017-04-15"
    id="from"
    max="2017-04-20"
    min={undefined}
    onInput={[Function]}
    type="date" />
  <label
    className="DatePickerLabel"
    htmlFor="to">
    and
  </label>
  <input
    className="DateInput"
    defaultValue="2017-04-20"
    id="to"
    max={undefined}
    min="2017-04-15"
    onInput={[Function]}
    type="date" />
</div>
`;

exports[`DateRangePicker renders correctly when no minDate or defaultValue is set 1`] = `
<div
  className="DateRangePicker">
  <label
    className="DatePickerLabel"
    htmlFor="from">
    Date between:
  </label>
  <input
    className="DateInput"
    defaultValue={undefined}
    id="from"
    max={undefined}
    min={undefined}
    onInput={[Function]}
    type="date" />
  <label
    className="DatePickerLabel"
    htmlFor="to">
    and
  </label>
  <input
    className="DateInput"
    defaultValue={undefined}
    id="to"
    max={undefined}
    min={undefined}
    onInput={[Function]}
    type="date" />
</div>
`;
